drivers/kw2xrf/kw2xrf_spi.c

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (C) 2015555555555555555555555555555555555555555555555555555555 PHYTEC Messtechnik GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     drivers_kw2xrf
+ * @{
+ * @file        kw2xrf_spi.c
+ * @brief       Implementation of SPI-functions for the kw2xrf driver
+ *
+ * @author      Johann Fischer <j.fischer@phytec.de>
+ * @author      Jonas Remmert <j.remmert@phytec.de>
+ * @}
+ */
+#include "kw2xrf.h"
+#include "kw2xrf_reg.h"
+#include "kw2xrf_spi.h"
+#include "kw2xrf_internal.h"
+
+#include "cpu-conf.h"
+#include "periph/gpio.h"
+#include "irq.h"
+
+/* SPI1 connected to the modem (ieee802.15.4 radio) */
+#define KW2XDRF_SPI                   SPI1
+#define KW2XDRF_SPI_CLKEN()           (SIM->SCGC6 |= (SIM_SCGC6_SPI1_MASK))
+#define KW2XDRF_SPI_CLKDIS()          (SIM->SCGC6 &= ~(SIM_SCGC6_SPI1_MASK))
+#define KW2XDRF_SPI_IRQ               SPI1_IRQn
+#define KW2XDRF_SPI_IRQ_HANDLER       isr_spi1
+
+void kw2xrf_spi_init(void)
```

Comment:
Well I'd guess that's the crux with hardware abstraction, it's hard to match all platforms' specialties. Some remarks

1. I don't see any logic in here which can not be used with the current SPI interface and mapped by macros. (If the Freescale SPI can be used without external CS I'm fine with it. Using GPIOs is no requirement)
2. If there is an interface for some functionality I think we should use it to minimise clutter. (`periph/spi.h` is always open for discussion, and I'd be happy if we find extensions to it while keeping it simple.)
3. In this current version the driver is not portable to other base MCUs and SPI specific code in different locations.
4. In my humble opinion I don't see advantages for HW CS if not used with DMA, especially in combination with turned off interrupts while busy waiting for transfers to finish.

To sum up, I'd very welcome if you could reconsider your decisions.